add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/EngineBindings.kt

```diff
@@ -0,0 +1,82 @@
+package dev.flutter.multipleflutters
+
+import android.app.Activity
+import io.flutter.FlutterInjector
+import io.flutter.embedding.engine.FlutterEngine
+import io.flutter.embedding.engine.dart.DartExecutor
+import io.flutter.plugin.common.MethodChannel
+
+/**
+ * This interface represents the notifications an EngineBindings may be receiving from the Flutter
+ * instance.
+ *
+ * What methods this interface has depends on the messages that are sent over the EngineBinding's
+ * channel in `main.dart`.  Messages that interact with the DataModel are handled automatically
+ * by the EngineBindings.
+ *
+ * @see main.dart for what messages are getting sent from Flutter.
+ */
+interface EngineBindingsDelegate {
+    fun onNext()
+}
+
+/**
+ * This binds a FlutterEngine instance with the DataModel and a channel for communicating with that
+ * engine.
+ *
+ * Messages involving the DataModel are handled by the EngineBindings, other messages are forwarded
+ * to the EngineBindingsDelegate.
+ *
+ * @see main.dart for what messages are getting sent from Flutter.
+ */
+class EngineBindings(activity: Activity, delegate: EngineBindingsDelegate, entrypoint: String) : DataModelObserver {
+    var channel: MethodChannel
+    var engine: FlutterEngine
+    val delegate: EngineBindingsDelegate
+
+    init {
+        val app: App = activity.applicationContext as App
+        // This has to be lazy to avoid creation before the FlutterEngineGroup.
+        val dartEntrypoint by lazy {
+            DartExecutor.DartEntrypoint(FlutterInjector.instance().flutterLoader().findAppBundlePath(), entrypoint)
+        }
+        engine = app.engines.createAndRunEngine(activity, dartEntrypoint)
+        this.delegate = delegate
+        channel = MethodChannel(engine.dartExecutor.binaryMessenger, "multiple-flutters")
+    }
+
+    /**
+     * This setups the messaging connections on the platform channel and the DataModel.
+     */
+    fun attach() {
+        DataModel.instance.addObserver(this)
+        channel.invokeMethod("setCount", DataModel.instance.counter)
```

Comment:
somewhat surprised this didn't ask you to null check it if it's a var

Reply:
Hmm, me too.